src/agent/onefuzz-supervisor/src/setup/double.rs

```diff
@@ -7,12 +7,16 @@ use super::*;
 pub struct SetupRunnerDouble {
     pub ran: Vec<WorkSet>,
     pub script: SetupOutput,
+    pub error_message: Option<String>,
 }
 
 #[async_trait]
 impl ISetupRunner for SetupRunnerDouble {
     async fn run(&mut self, work_set: &WorkSet) -> Result<SetupOutput> {
         self.ran.push(work_set.clone());
+        if let Some(error) = self.error_message.clone(){
+            anyhow::bail!(error);
```

Comment:
👌  excellent, I was hoping something like this would be easy to add.

Reply:
(I also like how this introduces fault injection at the level of the setup runner itself, vs. just the child)